stubs/psycopg2/psycopg2/_psycopg.pyi

```diff
@@ -478,24 +530,50 @@ class lobject:
     def unlink(self): ...
     def write(self, str): ...
 
-def Date(year, month, day): ...
-def DateFromPy(*args, **kwargs): ...
-def DateFromTicks(ticks): ...
-def IntervalFromPy(*args, **kwargs): ...
-def Time(hour, minutes, seconds, tzinfo=...): ...
-def TimeFromPy(*args, **kwargs): ...
-def TimeFromTicks(ticks): ...
-def Timestamp(year, month, day, hour, minutes, seconds, tzinfo=...): ...
-def TimestampFromPy(*args, **kwargs): ...
-def TimestampFromTicks(ticks): ...
+@type_check_only
+class _datetime:
```

Comment:
This and other classes that implement the `ISQLQuote` protocol could be made generic and their read-only `adapted` attribute would have the type variable as a type, shall someone ask for this later, but for now this is good enough.

Reply:
Maybe add this as a `# TODO` comment?